File: client/java/src/avt/Dog.java
package avt;

public final class Dog extends Animal {
   public static boolean r = false;
   public static AvPosition posDosTr;
   public static byte numBer = 0;
   public static short itemID = -1;

   public Dog() {
   }

   public Dog(int var1, byte var2) {
      super(0, 0, var1, var2);
      ++numBer;
   }

   public final void setInit() {
      this.setPos((FarmScr.numTileBarn + 3) * 24 + (CRes.rnd((LoadMap.wMap - FarmScr.numTilePond - FarmScr.numTileBarn - 5) * 6) << 2), 48 + (CRes.rnd(30) << 2));
   }

   public final void updateEat() {
      if (itemID == -1) {
         super.isEat = false;
      } else if (super.hunger && !super.isEat) {
         super.isEat = true;
      }

   }

   public final void updatePos() {
      super.posNext = new AvPosition();
      this.setPos();
   }

   public final void setPos() {
      AvPosition var2;
      if (super.isEat) {
         super.v = 2;
         var2 = posDosTr;
         super.posNext = var2;
      } else {
         var2 = new AvPosition(288 + (CRes.rnd(126) << 2), 24 + (CRes.rnd(36) << 2));
         super.posNext = var2;
      }

   }

   public final void reset() {
      if (!super.isEat && CRes.random(2) == 0) {
         super.cycle = 200;
      }

      if (super.isEat && CRes.distance(posDosTr.x, posDosTr.y, super.x, super.y) < 18) {
         super.isEat = false;
         super.hunger = false;
         super.cycle = 200;
         FarmScr.gI();
         FarmScr.doEat(itemID, super.IDDB);
      }

      super.reset();
   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/Dog.cs
public class Dog : Animal
{
	private class IActionBitten : IAction
	{
		public void perform()
		{
			FarmScr.gI().onBittenByDog();
			Canvas.endDlg();
		}
	}

	public static bool isHound;

	public static bool isCan;

	public static AvPosition posDosTr;

	public static sbyte numBer;

	public static short itemID = -1;

	public Dog()
	{
	}

	public Dog(int id, sbyte species)
		: base(0, 0, id, species)
	{
		numBer++;
	}

	public override void setInit()
	{
		setPos((FarmScr.numTileBarn + 3) * 24 + setX(), 48 + CRes.rnd(30) * 4);
	}

	public int setX()
	{
		int num = LoadMap.wMap - FarmScr.numTilePond - FarmScr.numTileBarn - 5;
		return CRes.rnd(num * 6) * 4;
	}

	public override void updateEat()
	{
		if (isHound && FarmScr.idFarm != GameMidlet.avatar.IDDB)
		{
			if (v < 3)
			{
				reset();
			}
			if (!isCan && CRes.abs(GameMidlet.avatar.x - x) <= 24 && CRes.abs(GameMidlet.avatar.y - y) <= 24)
			{
				isCan = true;
				Canvas.startOK(T.youAreBittenByDog, new IActionBitten());
				Canvas.msgdlg.setDelay(20);
			}
		}
		if (itemID == -1)
		{
			isEat = false;
		}
		else if (hunger && !isEat)
		{
			isEat = true;
		}
	}

	public override void updatePos()
	{
		posNext = new AvPosition();
		setPos();
	}

	public override void setPos()
	{
		if (isHound && FarmScr.idFarm != GameMidlet.avatar.IDDB)
		{
			if (health > 20 && !hunger)
			{
				v = 8;
			}
			else
			{
				v = 6;
			}
			setPosNext(new AvPosition(GameMidlet.avatar.x, GameMidlet.avatar.y));
		}
		else if (isEat)
		{
			v = 2;
			setPosNext(posDosTr);
		}
		else
		{
			setPosNext(new AvPosition(288 + CRes.rnd(126) * 4, 24 + CRes.rnd(36) * 4));
		}
	}

	public override void reset()
	{
		if (!isEat && CRes.random(2) == 0)
		{
			cycle = 200;
		}
		if (isEat && CRes.distance(posDosTr.x, posDosTr.y, x, y) < 18)
		{
			isEat = false;
			hunger = false;
			cycle = 200;
			if (FarmScr.gI().doEat(itemID, IDDB))
			{
				itemID = -1;
			}
		}
		base.reset();
		if (isHound && FarmScr.idFarm != GameMidlet.avatar.IDDB)
		{
			cycle = 0;
		}
	}
}
